### PR TITLE
[prometheus-druid-exporter] Added druid exporter

### PR DESCRIPTION
**Pull request by iamabhishek-dubey**
Signed-off-by: iamabhishek-dubey <abhishekbhardwaj510@gmail.com>

#### What this PR does / why we need it:

Adding a new chart `prometheus-druid-exporter` to `prometheus-community`

#### Which issue this PR fixes

NA

#### Special notes for your reviewer:

NA

#### Checklist

- [X] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by gkarthiks** [comment]
@iamabhishek-dubey Thanks for adding the druis-exporter and help growing the community. Since its a new chart, its easy and nice to have the artifacts annotations added as well to have an elegant display in artifacthub.io.

Example ref the PR here by @scottrigby https://github.com/prometheus-community/helm-charts/pull/95

**Comment by iamabhishek-dubey** [comment]
@gkarthiks Thanks for the feedback. I have incorporated the suggested changes. Please review

**Comment by scottrigby** [comment]
@gkarthiks we should add that to https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md. I'm afk for the weekend but can add that next week if someone doesn't beat me to it 🙂 

After seeing how it renders in artifact hub (this is a new feature after all), I may actually modify that Chart.yaml file:

- Since `home` points to the upstream project anyway, remove it from `sources` and `annotations. artifacthub.io/links`. Note that for some charts the project home link has a separate end user facing website than its source code, but in these cases it doesn't.

File: charts/prometheus-druid-exporter/Chart.yaml
Comment (scottrigby) [inline charts/prometheus-druid-exporter/Chart.yaml:27]: In order for this annotation to display in artifact hub the link must be in the `sources` list (it matches by URL).

But also see this comment I made just now about simplifying what I would have done earlier in retrospect https://github.com/prometheus-community/helm-charts/pull/114#issuecomment-695335157

File: charts/prometheus-druid-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-druid-exporter/README.md:63]: We should change any relative links in the README to full URLs to the main branch. See https://github.com/prometheus-community/helm-charts/issues/61 (note anchors used in other links are fine, so I didn't comment on those).

**Review by scottrigby — CHANGES_REQUESTED** [review]
Hey thanks, just a few changes

File: charts/prometheus-druid-exporter/README.md
Comment (gkarthiks) [inline charts/prometheus-druid-exporter/README.md:85]: Please change the link for `prometheus-operator` to https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack and rename as `kube-pometheus-stack`

**Comment by iamabhishek-dubey** [comment]
@gkarthiks and @scottrigby please review

**Comment by iamabhishek-dubey** [comment]
@gkarthiks , @scottrigby is there anything else needs to change?

**Comment by iamabhishek-dubey** [comment]
Guys, this PR is in the open state from 1 week, please let me know if some changes are required

**Review by gkarthiks — APPROVED** [review]
/lgtm

**Comment by gkarthiks** [comment]
sorry @iamabhishek-dubey. I was afk for the last week.

File: charts/prometheus-druid-exporter/Chart.yaml
Comment (scottrigby) [reply charts/prometheus-druid-exporter/Chart.yaml:25]: To clarify, in order for the Artifact Hub links annotation to overwrite the link name, the URL must already exist in the `sources` list. But now the sources list has been removed.

In fact, I may even suggest adding a direct link to the chart directory within this github repo as the source. It would be `https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-druid-exporter`

File: charts/prometheus-druid-exporter/README.md
Comment (scottrigby) [reply charts/prometheus-druid-exporter/README.md]: 👍

**Comment by scottrigby** [comment]
I clarified here https://github.com/prometheus-community/helm-charts/pull/114#discussion_r495567375

**Comment by iamabhishek-dubey** [comment]
Hey @scottrigby ,
Please review the PR

File: charts/prometheus-druid-exporter/Chart.yaml
Comment (scottrigby) [reply charts/prometheus-druid-exporter/Chart.yaml:27]: Looks like you've got it

**Comment by scottrigby** [comment]
@iamabhishek-dubey looks good. Got future ref, https://github.blog/changelog/2019-02-21-re-request-review-on-a-pull-request/

**Review by scottrigby — APPROVED** [review]

**Review by scottrigby — CHANGES_REQUESTED** [review]
Actually there is one more thing. You'll also need to add yourself to the `/.github/CODEOWNERS` file in this PR, or you won't be able to maintain the chart.

**Comment by iamabhishek-dubey** [comment]
@scottrigby thanks for that cool option for re-requesting review :D

**Comment by iamabhishek-dubey** [comment]
@scottrigby if everything looks good, please merge this PR :)

**Comment by iamabhishek-dubey** [comment]
Please review and merge this PR

File: .github/CODEOWNERS
Comment (scottrigby) [inline .github/CODEOWNERS:17]: This path needs to end with a trailing slash like the others:
```suggestion
/charts/prometheus-druid-exporter/ @iamabhishek-dubey
```

**Review by scottrigby — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (scottrigby) [reply .github/CODEOWNERS]: I went ahead and made this change, in an effort to get this out

**Comment by scottrigby** [comment]
@iamabhishek-dubey BTW will you be deprecating the chart at https://github.com/opstree/druid-exporter in favor of this once merged?

**Comment by scottrigby** [comment]
Interestingly CI failed with this error:
> wget: bad address 'prometheus-druid-exporter-jixxey9256:8080'

I re-ran the tests, but saw the same error.

The commit I made is entirely unrelated 81d1ddb9d827e110930d70c7c32adb9620e23104, so I'm not sure why it was succeeding before but failing now. Any thoughts?

**Comment by torstenwalter** [comment]
@scottrigby I re-ran the tests once more and they are green. I don't know why could a flaky test. I suggest to merge it now. If it happens again the maintainers can easily look into it.

**Review by torstenwalter — APPROVED** [review]

**Review by scottrigby — APPROVED** [review]

**Comment by scottrigby** [comment]
Thanks everyone for the patience and the work 👍

**Pull request merged**
